src/m365/cli/commands/config/config-set.spec.ts

```diff
@@ -0,0 +1,93 @@
+import * as assert from 'assert';
+import * as chalk from 'chalk';
+import * as sinon from 'sinon';
+import { Logger } from '../../../../cli';
+import Command from '../../../../Command';
+import commands from '../../commands';
+import configstore from '../../../../configstoreOptions';
+const command: Command = require('./config-set');
+
+describe(commands.CONFIG_SET, () => {
+  let log: any[];
+  let logger: Logger;
+  let loggerLogSpy: sinon.SinonSpy;
+  let loggerLogToStderrSpy: sinon.SinonSpy;
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: (msg: string) => {
+        log.push(msg);
+      }
+    };
+    
+    loggerLogSpy = sinon.spy(logger, 'log');
+    loggerLogToStderrSpy = sinon.spy(logger, 'logToStderr');
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name.startsWith(commands.CONFIG_SET), true);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it(`sets ${configstore.showHelpOnFailure} property`, (done) => {
```

Comment:
So if I understand that correctly, you add `stub().callsFake()` to replace the command's normal behavior by a stub, so that when running `command.action` it will run this stub instead, right?

Shouldn't it be restored with `stub.restore()`?

Reply:
Correct, fake replaces command behavior with whatever you define.

Typically, you would want to restore the original behavior, like you mentioned, but in this case, we're doing it in one place after all tests have been run: https://github.com/pnp/cli-microsoft365/blob/fd082c3a625d8466675b8b3d407aafd353a57546/src/m365/cli/commands/config/config-set.spec.ts#L30